qadence_protocols/mitigations/readout.py

```diff
@@ -7,9 +7,34 @@
 from numpy.linalg import inv, matrix_rank, pinv
 from qadence import QuantumModel
 from qadence.noise.protocols import Noise
-from qadence.types import ReadOutOptimization
 from scipy.linalg import norm
 from scipy.optimize import LinearConstraint, minimize
+from scipy.sparse import csr_matrix
+from scipy.sparse.linalg import gmres
+
+from qadence_protocols.types import ReadOutOptimization
+
+
+def subspace_kron(
+    noise_matrices: npt.NDArrary, subspace: npt.NDArray, normalize: bool = True
+) -> npt.NDArray:
+    n_qubits = len(noise_matrices)
+    conf_matrix = csr_matrix((2**n_qubits, 2**n_qubits))
+
+    for j in subspace:
+        for i in subspace:
+            bin_i = bin(i)[2:].zfill(n_qubits)
+            bin_j = bin(j)[2:].zfill(n_qubits)
+
+            ## manually computing the entries of tensor product for only the subspace
+            conf_matrix[i, j] = np.prod(
+                [noise_matrices[k][int(bin_i[k])][int(bin_j[k])] for k in range(n_qubits)]
+            )
+
+        if normalize:
```

Comment:
Are there cases where you _don't_ want to normalise ?

Reply:
I wanted the function to a be a standalone for just doing subspace kronecker, but didnt want to shift the normalizing outside. Leaving it in this fashion so that anyone who isnt interested in the normalization part can continue to use it. Let me know if this is not a good practice.

Reply:
Well, what I understood from my brief overview of the paper is that you always renormalise your subspace. I was just wondering if it made sense to leave the option open.

Reply:
True, just wanted to keep it for someone who is interested in using kronecker for a subspace. Maybe I will rename the function to reflect it then and remove the optional argument.